marketplace-builder/src/builder.rs

```diff
@@ -565,75 +599,74 @@ mod test {
-                        format!(
-                            "bundle_info/bundle/{parent_view_number}/{parent_commitment}/{}",
-                            parent_view_number + 1
-                        )
-                        .as_str(),
-                    )
-                    .send()
-                    .await
-                    .unwrap();
-                assert_eq!(bundle.transactions, vec![registered_transaction.clone()]);
-
-                let txn_commit = <[u8; 32]>::from(registered_transaction.commit()).to_vec();
-                let signature = bundle.signature;
-                assert!(signature.verify(txn_commit, address).is_ok());
-
-                let (payload, _) = Payload::from_transactions(
-                    vec![registered_transaction],
-                    &ValidatedState::default(),
-                    &NodeState::default(),
-                )
-                .await
-                .expect("unable to create payload");
+        let txn_commit = <[u8; 32]>::from(registered_transaction.commit()).to_vec();
+        let signature = bundle.signature;
+        assert!(signature.verify(txn_commit, address).is_ok());
 
-                let encoded_txns = payload.encode().to_vec();
-                let block_size = encoded_txns.len() as u64;
+        let (payload, _) = Payload::from_transactions(
+            vec![registered_transaction],
+            &ValidatedState::default(),
+            &NodeState::default(),
+        )
+        .await
+        .expect("unable to create payload");
 
-                let fees = base_fee * block_size;
+        let encoded_txns = payload.encode().to_vec();
+        let block_size = encoded_txns.len() as u64;
 
-                let fee_signature = <<SeqTypes  as NodeType>::BuilderSignatureKey as BuilderSignatureKey>::sign_sequencing_fee_marketplace(
-                    &keypair,
-                    fees.as_u64().unwrap(),
-                )
-                .unwrap();
+        let fees = base_fee * block_size;
 
-                let sequencing_fee = BuilderFee {
-                    fee_amount: fees.as_u64().unwrap(),
-                    fee_account: FeeAccount::from(address),
-                    fee_signature,
-                };
+        let fee_signature = <<SeqTypes  as NodeType>::BuilderSignatureKey as BuilderSignatureKey>::sign_sequencing_fee_marketplace(
+            &keypair,
+            fees.as_u64().unwrap(),
+        )
+        .unwrap();
 
-                assert_eq!(bundle.sequencing_fee, sequencing_fee);
+        let sequencing_fee = BuilderFee {
+            fee_amount: fees.as_u64().unwrap(),
+            fee_account: FeeAccount::from(address),
+            fee_signature,
+        };
 
-                break;
-            }
-        }
+        assert_eq!(bundle.sequencing_fee, sequencing_fee);
     }
 
-    #[async_std::test]
-    async fn test_marketplace_fallback_builder() {
+    async fn test_marketplace_fallback_builder(public_mempool: bool) {
```

Comment:
I think it would be better to rewrite this test to replace the `bool` in this test with an `enum` that can add clarity to each of these cases.  Since both forks end with an `Event` stream in either case, it would also be a natural place for the conditional to exist instead of within the body of this function.

Reply:
> Since both forks end with an `Event` stream in either case, it would also be a natural place for the conditional to exist instead of within the body of this function.

Could you clarify this? Besides replacing the `bool` with an `enum`, are you suggesting some other change?

Reply:
All make sense! I replaced `bool` with `enum` and reorganized functions. There are some parts that can't be combined or extracted easily, e.g., due to the public and private mempools using different event streams, but I added some helper functions to simplify things.